src/js/bun/sql.ts

```diff
@@ -228,6 +241,526 @@ init(
   },
 );
 
+function onQueryFinish(onClose) {
+  this.queries.delete(onClose);
+  this.pool.release(this);
+}
+
+enum PooledConnectionState {
+  pending = 0,
+  connected = 1,
+  closed = 2,
+}
+enum PooledConnectionFlags {
+  /// canBeConnected is used to indicate that at least one time we were able to connect to the database
+  canBeConnected = 1 << 0,
+  /// reserved is used to indicate that the connection is currently reserved
+  reserved = 1 << 1,
+  /// preReserved is used to indicate that the connection will be reserved in the future when queryCount drops to 0
+  preReserved = 1 << 2,
+}
+class PooledConnection {
+  pool: ConnectionPool;
+  connection: ReturnType<typeof createConnection>;
+  state: PooledConnectionState = PooledConnectionState.pending;
```

Comment:
technically the flags and the connection state can be the same property since the connection state is 2 bits and the flags are 3 bits and we can go up to 32 bit int

Reply:
but its probably fine